src/pages/Embeds/Plio.vue

```diff
@@ -1064,7 +1070,24 @@ export default {
         player_time: this.player.currentTime != null ? this.player.currentTime : 0,
         session: this.sessionDBId,
       };
-      EventAPIService.createEvent(eventData);
+      let response = await EventAPIService.createEvent(eventData);
+      if (eventType == "watching") this.watchingEventDBId = response.id;
+    },
+    /**
+     * Updates an event
+     * @param {String} eventType - The type of event that needs to be logged
+     * @param {Number} eventDBId - The id of an event as pulled from the database which needs to be updated
+     * @param {Object} eventDetails - An object with the details of the event
+     */
+    updateEvent(eventType, eventDBId, eventDetails = {}) {
+      if (!this.hasSessionStarted || !this.isAuthenticated || this.previewMode) return;
```

Comment:
this is not needed as the function invoking this function already places these checks.